src/levanter/checkpoint.py

```diff
@@ -348,7 +348,7 @@ def tree_deserialise_leaves(
     path: PathLike,
     like: PyTree,
     filter_spec=default_deserialise_filter_spec,
-    is_leaf: Callable[[Any], bool] = _is_index,
+    is_leaf: Optional[Callable[[Any], bool]] = None,
```

Comment:
Changes in this file are because the experimental `StateIndex` has been removed from Equinox. (It's now non-experimental and requires no special-casing.)

Reply:
it was cargo-cult anyhow.